packages/web-components/src/components/cbp-app-header/cbp-app-header.tsx

```diff
@@ -7,9 +7,46 @@ import { Component, Host, h, Prop } from '@stencil/core';
 export class CbpAppHeader {
 
   
+  private navItems: HTMLCbpNavItemElement[] = [];
+
+  @Element() host: HTMLElement;
+
   /** Specifies the context of the component as it applies to the visual design and whether it inverts when light/dark mode is toggled. Default behavior is "light-inverts" and does not have to be specified. */
   @Prop({ reflect: true }) context: "light-inverts" | "light-always" | "dark-inverts" | "dark-always";
 
+
+
+  initNavItemset() {
+    // check for a default navItem, otherwise set the first one active
+    let activeNavItem = this.navItems[0];
+    this.navItems.forEach(navItem => {
+      if (navItem.selected === true) activeNavItem = navItem;
+    });
+    this.setActiveNav(activeNavItem);
+  }
+
+  setActiveNav(activatedNav) {
+    this.navItems.forEach((navItem: HTMLCbpNavItemElement) => {
+      let link = navItem.querySelector('a');
```

Comment:
This function is accepting a node that is marked as selected, so you should handle buttons as well (regardless of other downstream integration concerns). If the dev set it to selected, it's selected. you can qs multiple elements easily enough: 

```
navItem.querySelector('a, button')
```

Reply:
updated as requested